### ISSUE TITLE
Unable to initialize local node: Time out while retrieving initial list of objects from kvstore

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#7248]
@loverto reports the following with Cilium v1.4.1:

```
level=info msg="New endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=349 identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="No request received to manage networking for container" containerID=3d0025b61f maxRetry=20 subsys=workload-watcher willRetry=false
level=info msg="No request received to manage networking for container" containerID=4ab5ad78c2 maxRetry=20 subsys=workload-watcher willRetry=false
level=info msg="No request received to manage networking for container" containerID=2568daeaa1 maxRetry=20 subsys=workload-watcher willRetry=false
level=info msg="Serving cilium health at http://[::]:4240" subsys=health-server
level=info msg="No request received to manage networking for container" containerID=8cafc0dd0f maxRetry=20 subsys=workload-watcher willRetry=false
level=info msg="Recompiled endpoint BPF program" BPFCompilationTime=6.450801672s containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=349 error="&lt;nil&gt;" identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Completed endpoint regeneration" bpfCompilation=6.450801672s buildDuration=6.463902673s containerID= datapathPolicyRevision=1 desiredPolicyRevision=1 endpointID=349 identity=4 ipv4= ipv6= k8sPodName=/ mapSync="699.489µs" policyCalculation="114.098µs" prepareBuild=2.202168ms proxyConfiguration="20.099µs" proxyPolicyCalculation=5.583815ms proxyWaitForAck="3.8µs" reason="health daemon bootstrap" subsys=endpoint waitingForCTClean="122.198µs" waitingForLock="4.9µs"
level=info msg="Serving cilium health at unix:///var/run/cilium/health.sock" subsys=health-server
level=info msg="Conntrack garbage collection statistics" completed=true duration=5.177222ms family=ipv4 maxEntries=1000000 numDeleted=5 numKeyFallbacks=0 numLookups=10 numLookupsFailed=0 protocol=TCP startTime="2019-02-28 09:17:29.765171451 +0000 UTC m=+78.141194864" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=6.193806ms family=ipv4 maxEntries=262144 numDeleted=0 numKeyFallbacks=0 numLookups=10 numLookupsFailed=0 protocol=non-TCP startTime="2019-02-28 09:17:29.770647268 +0000 UTC m=+78.146670681" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=4.375634ms family=ipv6 maxEntries=1000000 numDeleted=1 numKeyFallbacks=0 numLookups=2 numLookupsFailed=0 protocol=TCP startTime="2019-02-28 09:17:29.777155669 +0000 UTC m=+78.153179082" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=1.283081ms family=ipv6 maxEntries=262144 numDeleted=0 numKeyFallbacks=0 numLookups=2 numLookupsFailed=0 protocol=non-TCP startTime="2019-02-28 09:17:29.781824598 +0000 UTC m=+78.157848111" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=7.664884ms family=ipv4 maxEntries=1000000 numDeleted=5 numKeyFallbacks=0 numLookups=10 numLookupsFailed=0 protocol=TCP startTime="2019-02-28 09:18:29.78362613 +0000 UTC m=+138.159649543" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=1.818472ms family=ipv4 maxEntries=262144 numDeleted=5 numKeyFallbacks=0 numLookups=10 numLookupsFailed=0 protocol=non-TCP startTime="2019-02-28 09:18:29.79157891 +0000 UTC m=+138.167602323" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=11.038232ms family=ipv6 maxEntries=1000000 numDeleted=2 numKeyFallbacks=0 numLookups=3 numLookupsFailed=0 protocol=TCP startTime="2019-02-28 09:18:29.79355098 +0000 UTC m=+138.169574293" subsys=map-ct
level=info msg="Conntrack garbage collection statistics" completed=true duration=1.405179ms family=ipv6 maxEntries=262144 numDeleted=1 numKeyFallbacks=0 numLookups=2 numLookupsFailed=0 protocol=non-TCP startTime="2019-02-28 09:18:29.804909907 +0000 UTC m=+138.180933320" subsys=map-ct
level=fatal msg="Unable to initialize local node" error="Time out while retrieving initial list of objects from kvstore" subsys=daemon
```

@loverto can you provide the ConfigMap configuration you have? Did you also disable IPv4?

---
**Update**: Previous v1.4 releases (< v1.4.3) distributed `ConfigMap` examples with the wrong etcd TLS certificate paths. The details can be seen in the [PR #7350](https://github.com/cilium/cilium/pull/7350). For users affected by that issue, correcting the paths to match the latest examples from the documentation should resolve the issue: https://docs.cilium.io/en/stable/kubernetes/configuration/

More broadly, this error message indicates a kvstore connectivity issue which can manifest in a variety of situations.

**Comment by joestringer** [comment]
There appears to be some relation between this issue and the fix in #7350 , which should fix issues for future installations coming up (merged to master; docs being backported to v1.4 stable docs shortly). Still looking for additional information from any users that hit this.

**Comment by jrajahalme** [comment]
Same error message and cilium-agents in CrashLoop backoff. This when running flannel tests (IPv6 disabled) on the TPROXY PR. Since this happens before any network policy is running this should not be due to TPROXY redirections. 

https://jenkins.cilium.io/job/Cilium-PR-Flannel-hook/19/artifact/05aa1996_K8sPolicyTest_Basic_Test_Allows_traffic_with_k8s_default-allow_egress_policy.zip

Excerpt from pod-kube-system-cilium-q6m6r-cilium-agent-previous.log:
```
2019-03-10T18:22:23.861726703Z level=debug msg="Queueing container event" containerID=486ba84e6b event="exec_create: cilium status" subsys=workload-watcher
2019-03-10T18:22:23.8628329Z level=debug msg="Queueing container event" containerID=486ba84e6b event="exec_start: cilium status" subsys=workload-watcher
2019-03-10T18:22:23.980295644Z level=debug msg="Queueing container event" containerID=486ba84e6b event=exec_die subsys=workload-watcher
2019-03-10T18:22:24.041578589Z level=debug msg="Skipping CiliumEndpoint update because it has not changed" containerID=f67c904777 controller="sync-to-k8s-ciliumendpoint (2173)" datapathPolicyRevision=1 desiredPolicyRevision=1 endpointID=2173 identity=100 ipv4=10.244.1.34 ipv6= k8sPodName=kube-system/etcd-operator-7bc6d4fc6-4jl4d subsys=endpointsynchronizer
2019-03-10T18:22:24.041610744Z level=debug msg="Controller func execution time: 270.222µs" name="sync-to-k8s-ciliumendpoint (2173)" subsys=controller uuid=2bf7e583-4361-11e9-8018-080027051dad
2019-03-10T18:22:24.353927811Z level=debug msg="Controller func execution time: 239.592µs" name=sync-policymap-2173 subsys=controller uuid=2c2ca8e9-4361-11e9-8018-080027051dad
2019-03-10T18:22:24.443103615Z level=debug msg="custom dialer based on k8s service backend is dialing to \"cilium-etcd-client.kube-system.svc:2379\"" subsys=etcd
2019-03-10T18:22:24.833321961Z level=fatal msg="Unable to initialize local node" error="Time out while retrieving initial list of objects from kvstore" subsys=nodediscovery
2019-03-10T18:22:24.834790383Z level=fatal msg="Agent pipe unexpectedly closed, shutting down" subsys=cilium-node-monitor
```

**Comment by joestringer** [comment]
Note that the `error="Time out while retrieving initial list of objects from kvstore"` message is a common symptom of kvstore connectivity issues.

This bug appears to have been addressed by #7350. Please file a new issue or ask in Slack if you hit an issue like this.
